### PR TITLE
fix: update import of deprecated repository modules

### PR DESCRIPTION
**Pull request by b-kamphorst**
Depends on https://github.com/python-poetry/poetry/pull/6803.

### CODE REVIEW

File: src/poetry_plugin_export/exporter.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 from cleo.io.io import IO
 from poetry.core.packages.dependency_group import MAIN_GROUP
-from poetry.repositories.http import HTTPRepository
+from poetry.repositories.http_repository import HTTPRepository
```
Comment (radoering) [inline src/poetry_plugin_export/exporter.py:11]:
```suggestion
try:
    from poetry.repositories.http_repository import HTTPRepository
except ImportError:
    from poetry.repositories.http import HTTPRepository
```

File: tests/conftest.py [hunk 2]
```diff
@@ -13,8 +13,8 @@
 from poetry.core.packages.package import Package
 from poetry.factory import Factory
 from poetry.layouts import layout
-from poetry.repositories import Pool
 from poetry.repositories import Repository
+from poetry.repositories import RepositoryPool
```
Comment (radoering) [inline tests/conftest.py:17]:
```suggestion
try:
    from poetry.repositories import RepositoryPool
except ImportError:
    from poetry.repositories import Pool as RepositoryPool
```

**Review by radoering — CHANGES_REQUESTED** [review]
Can you make the imports forward compatible please? The scheme is try "new import" except "old import". Proposals below. You probably have to add some "type: ignore" or similar to satisfy mypy and flake8.

**Comment by b-kamphorst** [comment]
> Can you make the imports forward compatible please? The scheme is try "new import" except "old import". Proposals below. You probably have to add some "type: ignore" or similar to satisfy mypy and flake8.

Indeed, the current PR is valid only for `"poetry = ^1.3.0"` so that it is valid only after poetry 1.3.0 is released. I have made it backwards compatible in the way you describe so that we can release it already. If this PR is released only after the release of poetry 1.3.0, you may consider dropping the last commit (and require `poetry = ^1.3.0`).

**Comment by radoering** [comment]
Yeah, if poetry-plugin-export and poetry would not be so closely coupled we could just wait. Unfortunately, we are still delivering poetry-plugin-export with poetry. (We intend to stop doing this. Another maintainer has already created a proof-of-concept on how to proceed in a gracefully way but he hasn't been available lately.) That's why we want to release a version of poetry-plugin-export that can be delivered with poetry 1.3 and will not trigger a deprecation warning.

**Review by radoering — APPROVED** [review]

**Pull request merged**